src/msg/async/rdma/RDMAStack.cc

```diff
@@ -333,6 +336,22 @@ RDMAConnectedSocketImpl* RDMADispatcher::get_conn_lockless(uint32_t qp)
   return it->second.second;
 }
 
+Infiniband::QueuePair* RDMADispatcher::get_qp(uint32_t qp)
+{
+  Mutex::Locker l(lock);
+  // Try to find the QP in qp_conns firstly.
+  auto it = qp_conns.find(qp);
+  if (it == qp_conns.end()) {
```

Comment:
early return if found to reduce the indent level.

Reply:
Good catch:

if (it != qp_conns.end())
  return it->second.first;